nms/app/packages/magmalte/app/views/subscriber/SubscriberTable.js

```diff
@@ -0,0 +1,150 @@
+  const subscriberMetrics = ctx.metrics;
+  const [jsonDialog, setJsonDialog] = useState(false);
+  const [pageSize, setPageSize] = useState(10);
+  // first token (page 1) is an empty string
+  const [tokenList, setTokenList] = useState(['']);
+  const onClose = () => setJsonDialog(false);
+  const tableRef = React.useRef();
+
+  return (
+    <>
+      <div>
+        <JsonDialog open={jsonDialog} onClose={onClose} imsi={currRow.imsi} />
+        <ActionTable
+          tableRef={tableRef}
+          data={(query: ActionQuery) => {
+            return handleSubscriberQuery({
+              networkId,
+              query,
+              ctx,
+              tokenList,
+              setTokenList,
+              pageSize,
+              subscriberMetrics,
+            });
+          }}
+          columns={props.columns}
+          handleCurrRow={(row: SubscriberRowType) => setCurrRow(row)}
+          menuItems={
+            networkCtx.networkType === FEG_LTE
+              ? [
+                  {
+                    name: 'View JSON',
+                    handleFunc: () => {
+                      setJsonDialog(true);
+                    },
+                  },
+                ]
+              : [
+                  {
+                    name: 'View JSON',
+                    handleFunc: () => {
+                      setJsonDialog(true);
+                    },
+                  },
+                  {
+                    name: 'View',
+                    handleFunc: () => {
+                      history.push(relativeUrl('/' + currRow.imsi));
+                    },
+                  },
+                  {
+                    name: 'Edit',
+                    handleFunc: () => {
+                      history.push(relativeUrl('/' + currRow.imsi + '/config'));
+                    },
+                  },
+                  {
+                    name: 'Remove',
+                    handleFunc: () => {
+                      props
+                        .confirm(
+                          `Are you sure you want to delete ${currRow.imsi}?`,
+                        )
+                        .then(async confirmed => {
+                          if (!confirmed) {
+                            return;
+                          }
+
+                          try {
+                            await ctx.setState?.(currRow.imsi);
+                            tableRef.current?.onQueryChange();
+                          } catch (e) {
+                            enqueueSnackbar(
+                              'failed deleting subscriber ' + currRow.imsi,
+                              {
+                                variant: 'error',
+                              },
+                            );
+                          }
+                        });
+                    },
+                  },
+                ]
+          }
+          onChangeRowsPerPage={newPageSize => {
+            setPageSize(newPageSize);
```

Comment:
If it is easier not to give pageSize option, we can get rid of that. We can set it to say standard 25 subscribers per page and not have option to modify it. I'm not sure if people will modify that often